### PR TITLE
Refactor the touch logic and usage.

### PR DESCRIPTION
**Pull request by linfaxin**
1.The surface and bottom view could be clickable and longClickable, and it's selector drawables works fine;
2.The swipeLayout can wrap in other gestural layout, the swipe and other layout's gesture work's pretty;

### CODE REVIEW

**Comment by linfaxin** [comment]
@daimajia 
另外，我想 把 surfaceView 和 bottomView 必须为 ViewGroup 的限制去掉。
同时 所有 bottomView 支持在布局里用 layout_gravity 来标识自己的滑出方向（像DrawerLayout）
你怎么看 :)

**Comment by daimajia** [comment]
@linfaxin I'll check this PR later, thanks for your work on this project! ^.^ 

and i'm going to consider your idea seriously ( removing the constriction on the surfaceView and bottomView, and using layout_gravity to indicate the swipe direction) before replying you.

**Comment by linfaxin** [comment]
Improve the release drag logic, easy to open and close.

**Comment by JMPergar** [comment]
@linfaxin would be interesting that willOpenPercent will be customizable.

**Comment by linfaxin** [comment]
I see no SDK widget custom this. Maybe it's impossible?

**Comment by JMPergar** [comment]
On SwipeLayout.java

Line 1221:
    float willOpenPercent = (isCloseBeforeDragged ? mOpentPercentToAutomaticSwipe : 1 -
                mOpentPercentToAutomaticSwipe);

And:

    public float getOpentPercentToAutomaticSwipe() {
        return mOpentPercentToAutomaticSwipe;
    }

    public void setOpentPercentToAutomaticSwipe(float opentPercentToAutomaticSwipe) {
        mOpentPercentToAutomaticSwipe = opentPercentToAutomaticSwipe;
    }

    private float mOpentPercentToAutomaticSwipe = 0.25f;

**Comment by linfaxin** [comment]
Sorry, for my bad English. I mean it maybe not necessary.

**Comment by daimajia** [comment]
Hi, Mr @linfaxin .

Thanks your work on this project. Really sorry for my late reply cause recently was busy on some other heavy stuff. 

I'll renew this project, and try to update every day.  :-D

**Comment by daimajia** [comment]
@linfaxin 
I spent an hour to check your PR, it's awesome!  You made the code much cleaner.

File: library/src/main/java/com/daimajia/swipe/SwipeLayout.java [hunk 1]
```diff
@@ -1220,103 +1188,63 @@ public Status getOpenStatus() {
...
-                || mDragEdges.get(mCurrentDirectionIndex) == DragEdge.Right) {
-            if (xvel > 0) {
-                if (mDragEdges.get(mCurrentDirectionIndex) == DragEdge.Left)
-                    open();
+    protected void processHandRelease(float xvel, float yvel, boolean isCloseBeforeDragged) {
+        float minVelocity = mDragHelper.getMinVelocity();
+        View surfaceView = getSurfaceView();
+        DragEdge currentDragEdge = null;
+        try {
+            currentDragEdge = mCurrentDragEdge;
```
Comment (daimajia) [inline library/src/main/java/com/daimajia/swipe/SwipeLayout.java]: useless try catch

**Comment by linfaxin** [comment]
Wait your merge :)

**Comment by daimajia** [comment]
@linfaxin  May I add your wechat if you don't mind.

**Comment by linfaxin** [comment]
@daimajia I sent you a email :) please check

**Comment by eduplus** [comment]
+1 for this. I believe this might resolve the issues we have with the library right now. Hope this is merged and a new release build asap. Thanks @linfaxin and @daimajia

**Comment by daimajia** [comment]
@eduplus Please check the latest release v1.2.0 . I have pushed it to maven.

**Comment by eduplus** [comment]
@daimajia That fixed our issue. Great, thanks a ton.

**Comment by daimajia** [comment]
@eduplus Especially thanks @linfaxin for his pull request.  Enjoy it.

**Pull request merged**